### PR TITLE
Beautify readme

### PR DESCRIPTION
**Pull request by FirefoxMetzger**
Closes: https://github.com/pygfx/pygfx/issues/423

This PR updates the readme and makes it more presentable. It also adds the logo, which is used as favicon in the docs.

Todo:
- [x] finish refactoring later sections of the readme
- [x] add logo
- [x] make logo favicon in the sphinx docs

### CODE REVIEW

**Comment by FirefoxMetzger** [comment]
@Korijn Looking at the sections below the "under construction" note, I get the feeling that most of them should (and do) live somewhere in the docs already. I also don't think they are particularly relevant with the exception of the Introduction which highlights some features.

Are you okay with me liberally deleting the sections that I think won't add much value to new people finding the project and who have little familiarity with internals of rendering engines?

**Comment by Korijn** [comment]
Yes, absolutely. You can also consider breaking out parts to another file especially for contributors (e.g. the section about testing examples).

File: README.md
Comment (Korijn) [inline README.md:28]: Maybe also mention jupyter notebook setup?

Comment (almarklein) [inline README.md:28]: Since glfw is really light there is not a problem if its installed "extra", so I'd put less emphasis on the not-installing part. Something along the lines of:

```suggestion
The glfw library provides a lightweight window to render to. It's not strictly needed
if you only use pygfx from the notebook, use offscreen rendering, or plan to use 
another wgpu-compatible canvas such as PyQt, PySide, or wx.
```

**Comment by FirefoxMetzger** [comment]
@almarklein I am happy with the new readme for now. There is one discussion point on which features to highlight, but otherwise it should all be done.

RTR (and I will look into adding the logo in the mean time).

**Comment by Korijn** [comment]
Looks like the sponza lighting still needs some more tweaking:

image

File: README.md
Comment (Korijn) [inline README.md:28]: all the same caveats apply as for wgpu-py, e.g. you need pip >=20.3 on linux, etc. I think that's going to lead to the same kind of issues on this repo if we don't mention it like we do over there

File: README.md
Comment (Korijn) [inline README.md:109]: Are contributors smart enough to use a venv themselves?

**Comment by Korijn** [comment]
I'd recommend to add very prominent links to the User Guide, API Reference and Example Gallery near the top

File: README.md
Comment (Korijn) [inline README.md:67]: Let's make that "Builtin picking"

File: README.md
Comment (Korijn) [inline README.md:76]: I think the event system is also worth a mention

**Review by almarklein — COMMENTED** [review]
Perhaps we should also include a section on the goals / purpose. Something like the following (feel free to rewrite in your own words):

Pygfx is a graphics library that leverages the GPU (via wgpu-py). It can be used for classical graphics (e.g. meshes, PBR rendering, lighting, shadows, etc.) as well as scientific visualizations.

From the point of view of scientific visualization, pygfx provides an API that can be considered relatively low-level. This is a deliberate choice to achieve flexibility; higher-level API's can be built on top of it, such as [fastplotlib](https://github.com/kushalkolar/fastplotlib).

Using the GPU brings high performance, but this does not mean speed is our top priority; we favor hackability and correctness. If you want a top tier render engine, you should probably not use Python anyway 😉

File: README.md
Comment (almarklein) [reply README.md:76]: Idea: perhaps we can split out features particular of interest for 1) object rendering / meshes; 2) scientific viz.

File: README.md
Comment (FirefoxMetzger) [reply README.md:109]: I agree there are many existing ones, but I don't know a good one (never searched). If you know one I'd be happy to link it. It's mainly for people that are new to open-source work to have something easy to get them started.

Alternatively, we could drop the contributing section entirely or shrink it. Depends on how eager we are to attract external contributions.

File: README.md
Comment (FirefoxMetzger) [reply README.md:28]: I added a link to the platform requirements for wgpu-py and a general note.

File: README.md
Comment (FirefoxMetzger) [reply README.md:76]: I would only list the 3 or 5 features that we are most proud of (or that sell the best) otherwise it quickly becomes a boring cloth-line.

For all the others, we should already have a feature example that shows off each feature right? So we can just point users to the gallery and say "check if the feature you want is listed here".

**Comment by FirefoxMetzger** [comment]
@Korijn Can we get a version of the logo that combines the logo (on the left) and the name of the project (on the right)?

**Comment by Korijn** [comment]
> @Korijn Can we get a version of the logo that combines the logo (on the left) and the name of the project (on the right)?

I can request it but it won't be ready anytime soon :)

**Comment by FirefoxMetzger** [comment]
> I can request it but it won't be ready anytime soon :)

That's fine. As long as it is going through the pipeline we can replace it eventually ⭐ 

> I'll see if I can put something together myself later tonight. Any specifics like resolution I should take into account?

A landscape format would be good. If you want a specific solution, `600x300` is a good size. 

I also just realized that we can use svg in the readme, so I will kick out the png :)

**Comment by Korijn** [comment]
Here you go:

Banner

File: README.md
Comment (almarklein) [reply README.md:123]: This one perhaps? https://docs.github.com/en/get-started/quickstart/github-flow

**Comment by almarklein** [comment]
I tried to compose a "what is pygfx" that includes most of the features, which may be less tedious than a list of bullets. We may still cut some out for brevity though. Thoughts?

### What is pygfx

Pygfx is a graphics library that leverages the GPU via [WGPU](https://github.com/pygfx/wgpu-py) (the successor to OpenGL), and has an API heavily influenced by [ThreeJS](https://threejs.org/).

It can be used as a **render engine**, with support for physically based rendering, lights and shadows, order-independent transparenc, and more.

It also supports **scientific visualiations** with features like SDF text, colormaps, volume rendering, mesh- and volume slices, and the possibility to write custom shaders.

Interactive applications benefit from a powerful event system and picking mechanism.


### What it's not

From the point of view of scientific visualization, pygfx provides an API that can be considered relatively low-level. This is a deliberate choice to achieve flexibility; higher-level API's can be built on top of it, such as fastplotlib.

Using the GPU brings high performance, but this does not mean speed is our top priority; we favor hackability and correctness. If you want a top tier render engine, you should probably not use Python anyway 😉

File: README.md
Comment (FirefoxMetzger) [reply README.md:123]: Sounds good. I will still keep a snippet on how to get a dev install, because that is specific to pygfx (i.e. which optional dependencies to install).

**Comment by FirefoxMetzger** [comment]
> Here you go (two font size choices):

Thanks @Korijn . I found the default Inkscape font a bit simple, so I spend some time looking for a serif font that could work. This is what I've settled for, let me know if you prefer this one or if we should go with the logo you've shared.

pygfx_with_name

> I tried to compose a "what is pygfx" that includes most of the features, which may be less tedious than a list of bullets. We may still cut some out for brevity though. Thoughts?

I like it. I will put it into a "About pygfx" section which can live in the middle of the readme. I would keep this separate from the feature highlights though, because it is less of an "elevator pitch" and more of a "I'm hooked give me some details".

File: README.md
Comment (FirefoxMetzger) [reply README.md]: I've made a selection of the top 5 features from all our suggestions. I kicked out some (light/shaddows, PBR, ...) that you could already see in the example images at the top, or that is mentioned elsewhere (TreeJS/WGPU) and then choose the 5 items that sounded the most impressive.

Suggestions are, ofc, welcome, and we also have to find the corresponding examples for each feature.

**Comment by FirefoxMetzger** [comment]
Okay, this should resolve everything that was pending on this PR. A few things that need to be reviewed in this round:

- [ ] The logo that I added to the top of the readme
- [ ] The 5 feature highlights + links to examples
- [ ] the about section that I've added (+ maybe a citation for the project if we have/care)

Otherwise this PR should be ready.

File: README.md
Comment (almarklein) [inline README.md:32]:
```suggestion
offscreen or notebook rendering you may choose to omit glfw. Similarly, you can
```

File: README.md
Comment (almarklein) [inline README.md:70]:
```suggestion
- SDF based text rendering (example)
```

File: README.md
Comment (almarklein) [inline README.md:72]: The shader templating is cool, but only interesting for ppl that want to extend pygfx. Let's replace with lights etc.
```suggestion
- lights, shadows, and physically based rendering (PBR)
```

File: README.md
Comment (almarklein) [inline README.md:105]: The next sentence is more a consequence than a contradiction
```suggestion
[here](https://docs.github.com/en/get-started/quickstart/github-flow);
```

File: README.md
Comment (almarklein) [inline README.md:83]: It's more "inspired by" than "mimics", because we (deliberately) in some cases :)

File: README.md
Comment (almarklein) [inline README.md:85]: Can we flip this around - first mention it's general purpose, then say it's geared towards scientific vis?

File: README.md
Comment (almarklein) [inline README.md:91]: What kind of citation do you mean?

**Review by almarklein — COMMENTED** [review]
I made a few suggestions.

File: README.md
Comment (FirefoxMetzger) [reply README.md:85]: swapped

File: README.md
Comment (FirefoxMetzger) [reply README.md:82]: @ivoflipse This one is at the beginning of a new sentence (paragraph in fact) so I capitalized it. It feels strange not to capitalize, though we can ofc use lowercase if you prefer.

File: README.md
Comment (FirefoxMetzger) [reply README.md:86]: fixed

**Comment by Korijn** [comment]
Last one from me: `jupyter_rfb` is required for notebook usage, let's include that under "Installation"

File: README.md
Comment (ivoflipse) [reply README.md:87]: What does scikit-learn do for example?

Comment (ivoflipse) [reply README.md:87]: It seems to vary, which probably depends on who wrote it

https://scikit-learn.org/stable/faq.html

Comment (FirefoxMetzger) [reply README.md:87]: On second thought, let's avoid starting a long discussion over something minor. I'll change it.

File: README.md
Comment (almarklein) [reply README.md:96]: Let's skip for now.

File: README.md
Comment (almarklein) [inline README.md:78]: Consistency.
```suggestion
- texture and color mapping supporting 1D, 2D and 3D data ([example](
```

Comment (FirefoxMetzger) [reply README.md:78]: nice catch.

**Comment by FirefoxMetzger** [comment]
> Looks like the sponza lighting still needs some more tweaking:

Do we want to work out what tweaks sponza needs before we merge? 


If not, I think we are LGTM unless one of you knows a better example for a feature highlight compared to the one I chose.

**Comment by Korijn** [comment]
Nah, I'm good with merging at this point. Let's create follow up issues for any left overs.

**Comment by FirefoxMetzger** [comment]
@Korijn Shouldn't you be on vacation? 😅 We can let the issue sit here and you can merge it and we can deal with any consequences of that once you return.

File: README.md
Comment (almarklein) [inline README.md:38]: These platform requirements in wgpu-py have moved into its docs.

```suggestion
In addition there are some platform
requirements, see the [wgpu docs](https://wgpu-py.readthedocs.io/en/stable/start.html). In
essence, you need modern (enough) graphics drivers, and `pip>=20.3`.
```

**Comment by almarklein** [comment]
This is ready too @Korijn  :)

**Comment by FirefoxMetzger** [comment]
RTR :)

**Comment by Korijn** [comment]
Can you create an issue to track any remaining discussion items?

**Review by Korijn — APPROVED** [review]

**Comment by FirefoxMetzger** [comment]
@Korijn I think only the sponza scene situation is left. Otherwise I think we resolved everything that came up here.

**Pull request merged**
